### PR TITLE
[visit-struct] Update to 1.1.0 and prefix targets with unofficial 

### PR DESCRIPTION
**Pull request by traversaro**
This PR updates visit-struct to 1.1.0 after the recent tag: https://github.com/cbeck88/visit_struct/releases/tag/v1.1.0 . As I also noticed that the targets installed were not complaint with the mantainer guide, I also fixed those.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/visit-struct/portfile.cmake [hunk 1]
```diff
@@ -2,8 +2,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO cbeck88/visit_struct
-    REF v1.0
-    SHA512 1396d0d4c4d37f48434361d1e0ab4cb02c397aff1134678b26de713a27a4fcfa1c352890845502be645ba01e20314bf67731893fc6410b93e4521c1261d63c06
+    REF v1.1.0
```
Comment (jimwang118) [inline ports/visit-struct/portfile.cmake:5]:
```suggestion
    REF "v${VERSION}"
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/visit-struct/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "visit-struct",
-  "version": "1.0",
-  "port-version": 4,
+  "version": "1.1.0",
   "description": "A header-only library providing structure visitors for C++11 and C++14",
```
Comment (jimwang118) [inline ports/visit-struct/vcpkg.json:4]:
```suggestion
  "description": "A header-only library providing structure visitors for C++11 and C++14",
  "license": "BSL-1.0",
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
The usage has been tested successfully locally.
```
find_package(unofficial-visit_struct CONFIG REQUIRED)
target_link_libraries(main PRIVATE unofficial::visit_struct::visit_struct)
```

File: ports/visit-struct/vcpkg.json (hunk 2 above)
Comment (jimwang118) [reply ports/visit-struct/vcpkg.json:4]: > Done and rebased, thanks for the feedback @jimwang118 !

I checked that there is a license in the upstream, we need to add the keyword `"license": "BSL-1.0"` in `vcpkg.json`.

**Review by jimwang118 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**